src/continuum_opacity/opacity_H.jl

```diff
@@ -199,43 +198,91 @@ function Hminus_ff(nH_I_div_partition::Flt, ne::Flt, ν::Flt, ρ::Flt,
+from Table 1 of Bates (1952). Gray (2005) notes that the approximations match Bates' (1952)
+tabulated absorption data for 3800 Å ≤ λ ≤ 25000 Å at an accuracy of 0.3%. After reviewing the data
+in Bates (1952) I've concluded that Gray (2005) actually rounded down the lower end of the
+wavelength from 10⁵/26 Å or ∼3847 Å. (In other words, the approximation is good for
+3847 Å ≤ λ ≤ 25000.0 Å).
+
+While this may fit to σ₁ and U₁ to better than 0.3%, comparisons against values from table 2 of
+Bates (1952) indicate that the function reproduces the full absorption coeffient at better than
+1.5%. Gray (2005) did not provide a temperature range, but Bates (1952) only computed the
+absorption coefficient for 2500 K ≤ T ≤ 12000 K. In the text they have a comment that the proton's
+De Broglie wavelength is large at 2500 K and their semi-classical treatment may start to break
+down. However, as long as use n(H I, n=1), there doesn't seem to be any reason for us to enforce an
+upper Temperature limit.
+
+Bates (1952) states that his classical treatment of the interaction is probably most accurate at
+higher temperatures and longer wavelengths (note that the longest λ considered in the paper is 8
+times larger than the max λ the polynomials are fit against). He suggests that treatment is
+probably correct "to well within one part in ten even at the lower temperatures and [lower
+wavelengths]."
 """
-function H2plus_bf_and_ff(nH_I_div_partition::Float64, nHII::Float64, ν::Float64, ρ::Float64,
+function H2plus_bf_and_ff(nH_I_div_partition::Float64, nH_II::Float64, ν::Float64, ρ::Float64,
                           T::Float64)
-    error("This seems to be broken.")
-    # compute the number density of H I in the ground state
-    nHI_gs = ndens_state_hydrogenic(1, nH_I_div_partition, T, _H_I_ion_energy)
+    λ = c_cgs*1e8/ν # in ångstroms
+    if !(3846.15 <= λ <= 25000.0) # the lower limit is set to include 1.e5/26 Å
+        throw(DomainError(λ, "The wavelength must lie in the interval [3847 Å, 25000 Å]"))
+    end
+
+    if T < 2500  # we might be able to drop upper limit
+        throw(DomainError(T, "The temperature must be greater than or equal to 2500 K."))
+    end
+
+    β_eV = 1.0/(kboltz_eV * T)
+
+    # coef should be roughly 2.51e-42
+    coef = 16*π^4*bohr_radius_cgs^5*electron_charge_cgs^2/(3*hplanck_cgs*c_cgs) # cm⁵
+    stimulated_emission_correction = (1 - exp(-hplanck_eV*ν*β_eV))
+
+    logλ = log10(λ)
+    log2λ = logλ*logλ
+    log3λ = log2λ*logλ
+
+    atomic_cross_section = begin
+        σ1 = -1040.54 + 1345.71 * logλ - 547.628 * log2λ + 71.9684 * log3λ
+        # there was a typo in Gray (2005). In the book they give the following polynomial
+        # as the fit for U₁. In reality, it is the fit for negative U₁
+        neg_U1 = 54.0532 - 32.713 * logλ + 6.6699 * log2λ - 0.4574 * log3λ
+
+        # note: Gray (2005) used the equivalent expression: σ1*10^(neg_U1 * θ) where θ = 5040/T
+        σ1 * exp(neg_U1 * β_eV)
+    end
```

Comment:
This isn't important, but I used to like this pattern because I think it introduced a separate scope and that local variables would be captured.  Since you pointed out that this isn't true, I'm not sure what the benefits are.  IDK it's definitely not a problem to do it this way though.